catkin_ws/src/state_estimation/src/sensors.py

```diff
@@ -19,111 +18,121 @@
-        different = False
-        for i in range(len(current_state)):	
-            if isinstance(current_state[i], Iterable):
-                different = True if different else any([curr_state_el != last_state_el for curr_state_el, last_state_el in zip(current_state[i], self.last_state[i])])
-            else:
-                different = True if different else current_state[i] is not self.last_state[i]
-        
-        if different:
-            if rospy.get_time() - self.last_unique_state_time > self.time_before_considered_inactive:
-                rospy.loginfo("{} has become active.".format(self.sensor_name))
-            self.last_unique_state_time = rospy.get_time()
-            self.last_state = current_state
-        
-    def isActive(self):
-        if rospy.get_time() == 0: return False
-        if not self.hasValidData(): return False #check that state is complete
-        if rospy.get_time() - self.last_unique_state_time > self.time_before_considered_inactive: #check that state has changed in last N seconds
-            if rospy.get_time() - self.last_error_message_time > 1:
-                self.last_error_message_time = rospy.get_time()
-                rospy.logwarn("{} has been inactive for {} seconds.".format(self.sensor_name, self.time_before_considered_inactive))
-            return False
-        else:
-            return True
+    def get_is_active(self):
+        return self.is_active
 
-    def hasValidData(self):
-        raise NotImplementedError    
 
+# Depth Sensor class inheriting from Sensor
 class DepthSensor(Sensor):
     def __init__(self):
         super().__init__("Depth Sensor")
+        
+        self.z = 0
         self.z_pos_mount_offset = 0
+
+        rospy.Subscriber("/sensors/depth/status", Int32, self.sensor_status_cb)
         rospy.Subscriber("/sensors/depth/z", Float64, self.depth_cb)
 
-    def depth_cb(self, depth_msg):
-        self.z = depth_msg.data + self.z_pos_mount_offset
-        self.updateLastState()
-    
-    def hasValidData(self):
-        return self.z is not None
+    def depth_cb(self, msg):
+        self.z = msg.data + self.z_pos_mount_offset
+
+    def get_reading(self):
+        return {
+            "z": self.z
+        }
 
+
+# IMU class inheriting from Sensor
 class IMU(Sensor):
     def __init__(self):
         super().__init__("IMU")
-
+        
         q_imunominal_imu_w = rospy.get_param("q_imunominal_imu_w")
         q_imunominal_imu_x = rospy.get_param("q_imunominal_imu_x")
         q_imunominal_imu_y = rospy.get_param("q_imunominal_imu_y")
         q_imunominal_imu_z = rospy.get_param("q_imunominal_imu_z")
 
         self.q_imunominal_imu = np.quaternion(q_imunominal_imu_w, q_imunominal_imu_x, q_imunominal_imu_y, q_imunominal_imu_z)
         self.q_imu_auv = self.q_imunominal_imu.conjugate() * Q_IMUNOMINAL_AUV
-        self.q_nwu_auv = None
-        
+        self.q_nwu_auv = np.quaternion(1, 0, 0, 0)
+
+        self.angular_velocity = np.array([0,0,0])
+
```

Comment:
these are just the initial values because None was causing problems, and I didn't want to have to add if not None statements throughout the code